src/Apps/Order2/Routes/Checkout/CheckoutContext/Order2CheckoutContext.tsx

```diff
@@ -512,6 +513,42 @@ interface Order2CheckoutContextProviderProps {
   children: React.ReactNode
 }
 
+const NavigationGuardsManager: React.FC = () => {
+  const isLoading = Order2CheckoutContext.useStoreState(
+    state => state.isLoading,
+  )
+  const criticalCheckoutError = Order2CheckoutContext.useStoreState(
+    state => state.criticalCheckoutError,
+  )
+
+  useEffect(() => {
+    // Don't set up guards if there's a critical error or still loading
+    if (isLoading || criticalCheckoutError) {
+      return
+    }
+
+    window.addEventListener("beforeunload", preventHardReload)
+    window.history.pushState(null, "", window.location.pathname)
+    window.addEventListener("popstate", handleBackNavigation)
+
+    return () => {
+      window.removeEventListener("beforeunload", preventHardReload)
+      window.removeEventListener("popstate", handleBackNavigation)
+    }
+  }, [isLoading, criticalCheckoutError])
```

Comment:
This is very similar to the new code that prevents scroll during load in CheckoutLoadingManager.tsx.  maybe it should be in one place.